flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -3534,6 +3556,21 @@ class ArrayExprLowering {
   CC genarr(const Fortran::semantics::SymbolRef &sym) {
     return genarr(asScalarRef(sym));
   }
+  
+  /// Build an ExtendedValue from a fir.array<?x...?xT> without
+  /// actually setting the actual extents and lengths. This is only
+  /// to allow their propagation as ExtendedValue without triggering
+  /// verifier failure when propagating character/arrays as unbox value.
+  ExtValue abstractArrayExtValue(mlir::Value val) {
+    auto type = val.getType();
+    if (auto ty = fir::dyn_cast_ptrEleTy(type))
+      type = ty;
+    auto seqTy = type.cast<fir::SequenceType>();
+    llvm::SmallVector<mlir::Value> extents(seqTy.getDimension()); 
```

Comment:
I'm sort of curious how this helps propagate the LEN expression, since it appears to be using a nullptr.

FWIW, we've been burned with nullptr Value before, so I'm also wondering if we don't want a `fir::UndefOp` default value in the ctor call.